sql/catalyst/src/main/scala/org/apache/spark/sql/types/NullType.scala

```diff
@@ -32,6 +32,8 @@ class NullType private() extends DataType {
   override def defaultSize: Int = 1
 
   private[spark] override def asNullable: NullType = this
+
+  override def sql: String = "VOID"
```

Comment:
why not override `simpleString`? `sql` calls `simpleString`

Reply:
Just consider that simpleString/catalogString is a name of spark, and sql is used in other database ? We can still show something like `unknown` with `NullType`, If we can divide this.

I'm not sure the sql original meaning, but it seems used for Hive DDL e.g., `StructType.toDDL` use sql to build DDL string.

Reply:
It's confusing if `simpleString` and `sql` is not consistent.